compute_endpoint/tests/integration/endpoint/endpoint/test_gceengine_bad_worker_init.py

```diff
@@ -0,0 +1,121 @@
+import logging
+import uuid
+from queue import Queue
+
+import pytest
+from globus_compute_common import messagepack
+from globus_compute_endpoint.engines import GlobusComputeEngine
+from globus_compute_sdk.serialize import ComputeSerializer
+from parsl.executors.errors import BadStateException
+from parsl.jobs.errors import TooManyJobFailuresError
+from parsl.providers import LocalProvider
+from tests.utils import double, ez_pack_function
+
+logger = logging.getLogger(__name__)
+
+
+@pytest.fixture
+def gc_engine_bad_submit_command(tmp_path):
+    ep_id = uuid.uuid4()
+    engine = GlobusComputeEngine(
+        address="127.0.0.1",
+        heartbeat_period=1,
+        heartbeat_threshold=2,
+        provider=LocalProvider(
+            init_blocks=1,
+            min_blocks=0,
+            max_blocks=1,
+            worker_init='echo "BAD SUBMIT COMMAND"; exit 0',
+        ),
+        max_idletime=0,
+        strategy="simple",
+    )
+    queue = Queue()
+    engine.start(endpoint_id=ep_id, run_dir=str(tmp_path), results_passthrough=queue)
+
+    yield engine
+    engine.shutdown()
+
+
+def test_bad_submit_command(gc_engine_bad_submit_command, caplog):
+    engine = gc_engine_bad_submit_command
+    queue = engine.results_passthrough
+    caplog.set_level(logging.WARNING)
+
+    task_id = uuid.uuid4()
+    serializer = ComputeSerializer()
+    task_body = ez_pack_function(
+        serializer,
+        double,
+        (5,),
+        {},
+    )
+    task_message = messagepack.pack(
+        messagepack.message_types.Task(task_id=task_id, task_buffer=task_body)
+    )
+    future = engine.submit(task_id=task_id, packed_task=task_message)
+
+    with pytest.raises(BadStateException):
+        future.result(timeout=60)
```

Comment:
Just observed this timeout.  For the tests, in an environment that we control, can we reduce this timeout?  This is asking for test time bloat that I'm keen to avoid when we test locally.

Reply:
I got this down to 2s